src/lib/aqua.ts

```diff
@@ -52,27 +52,30 @@ const getAquaFilesRecursively = async (dirPath: string): Promise<string[]> => {
 
 const writeFileAndMakeSureDirExists = async (
   filePath: string,
-  data: string,
+  dataPromise: string | Promise<string | undefined>,
```

Comment:
this makes function too smart - it executes _any_ effectful callback. This function could operate just on `string | undefined` type.
```suggestion
  fileContent: string | undefined,
```

Reply:
In the code below, in flatMap callback, you can await result before passing it to this function.

Reply:
For some reason I thought this would be impossible, but looks like flatMap is smart enough when working with async functions as callbacks